props/prop_base.h

```diff
@@ -300,15 +316,103 @@ class PropBase : CommandParser, Looper, protected SaberBase {
   } while(0);
 
     ONCEPERBLADE(ACTIVATE);
+    #ifdef SAVE_STATE
+    ResumePreset();
+    #else
     SetPreset(0, false);
+    #endif
     return;
 
    bad_blade:
     STDOUT.println("BAD BLADE");
 #ifdef ENABLE_AUDIO
     talkie.Say(talkie_error_in_15, 15);
     talkie.Say(talkie_blade_array_15, 15);
-#endif    
+#endif
+  }
+
+  void ResumePreset() {
+    savestate_.ReadInCurrentDir("curstate.ini");
+    SetPreset(savestate_.preset, false);
+    if (savestate_.volume < VOLUME) {
+      dynamic_mixer.set_volume(savestate_.volume);
+    }
+  }
+
+  bool OpenState(FileReader* f, const char* filename) {
+    if (!f->Open(filename))
+      return false;
+
+    if (f->FileSize() < 4) return false;
+    int p = f->FileSize() - 1;
+    while (p > 0) { f->Seek(p); if (!isSpace(f->Read())) break; p--; }
+    f->Seek(p - 3);
+    if (!isSpace(f->Read())) return false;
+    if (toLower(f->Read()) != 'e') return false;
+    if (toLower(f->Read()) != 'n') return false;
+    if (toLower(f->Read()) != 'd') return false;
+    f->Seek(0);
+
+    return true;
+  }
+
+  bool SaveState(int preset) {
```

Comment:
I think too much stuff got copied from current_preset.h
For this file, I think all we really need is:

   save to curstate.tmp
   save to curstate.ini

If reading the ini file fails (or if the "end" is missing), we need to read the tmp file.
Sorry for making this so confusing.

Also, if there is code that we can re-use generalize and use in both places, we should do that as that uses less memory than simply copying and modifying.

Reply:
ok, pulled out the createini stuff and put a couple checks on the save state to try and open the ini, then the tmp.  Not sure about any code generalization right now but I can see a much larger effort to combine all the current preset and config file code into one base that reads and writes to generic ini files with key-pair variables.

Reply:
removed update ini and readstate, changed config_file.h to return a bool if it can't open a file.